pkg/controller/account/ec2.go

```diff
@@ -100,9 +100,16 @@ func (r *ReconcileAccount) BuildandDestroyEC2Instances(reqLogger logr.Logger, aw
 
 	// Wait till instance is running
 	var DescError error
+	totalWait := controllerutils.WaitTime * 60
+	currentWait := 1
 	for i := 0; i < 300; i++ {
+		currentWait = currentWait * 2
+		if currentWait > totalWait {
```

Comment:
I'm a bit confused by this logic. Could you explain what you're doing here?

Reply:
@fahlmant Oops, It is supposed to be `for totalWait > 0 {`. To put simply we double the time between checking if the EC2 instance is running `2 sec, 4 sec, 8 sec, 16 sec,...` until we reach 25 minutes. However, right now instead of ending the loop at 25 minutes it would reconcile for the rest of 300 loops, wouldn't crash but definitely bad. I'll add a comment and fix the loop. Thanks for catching!

Reply:
I think the other loop in this file has the same issue.

Reply:
No it doesn't